sys/arduino/serialport.cpp

```diff
@@ -41,19 +42,21 @@ SerialPort::SerialPort(uart_t dev)
 
 int SerialPort::available(void)
 {
-    return (int)rx_buf.avail;
+    return stdio_avail();
```

Comment:
This could be realized without extending the `stdio_base` API by using a ringbuffer as for direct used of `periph_uart`. In that case `Serial::begin` would clear the ringbuffer and `stdio_base` wouldn't be necessary.